provider/src/test/kotlin/org/gradle/kotlin/dsl/NamedDomainObjectCollectionExtensionsTest.kt

```diff
@@ -110,6 +112,21 @@ class NamedDomainObjectCollectionExtensionsTest {
         }
     }
 
+    @Test
+    fun `can access existing element by getting with reified type`() {
+        val element = DomainObject()
+        val container = mock<PolymorphicDomainObjectContainer<SomeObject>> {
+            on { getByName("domainObject") } doReturn element
+        }
+
+        container { // invoke syntax reified method
+            val domainObject by getting<DomainObject>()
+            assertThat(domainObject, sameInstance(element))
+        }
+
+        // regular syntax reified method won't have the same behaviour or compile time guarantees
```

Comment:
:o: This means reified syntax is not available for qualified collection references such as `val myTask by tasks.getting<MyTask>` which creates an asymmetry in the DSL that IIRC was the reason we didn't include it in the first round.

We also have to solve the related problem of not providing the _invoke syntax DSL_ when plugins declare configuration methods such as `plugins(configure: Action<PluginContainer>)`. We should consider this PR in light of that. 

@eskatos any thoughts?

Reply:
Would you like me to remove the reified `getting` methods?